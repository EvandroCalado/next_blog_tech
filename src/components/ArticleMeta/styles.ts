import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
