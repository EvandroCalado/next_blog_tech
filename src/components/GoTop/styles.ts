import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.primary};
    border: 1px solid ${theme.colors.primary};
    border-radius: 50%;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 5rem;
    height: 5rem;
    z-index: 6;
    text-decoration: none;
  `}
`;
