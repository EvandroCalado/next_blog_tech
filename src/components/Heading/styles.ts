import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: 900;
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: 900;
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: 900;
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
    font-weight: 900;
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primary : theme.colors.white};
    line-height: 1.2;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
    text-align: center;
  `}
`;
