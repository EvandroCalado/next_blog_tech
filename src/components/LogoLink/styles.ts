import styled from 'styled-components';

export const Container = styled.a`
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
  padding: 2rem;

  > img {
    width: 15rem;
    border-radius: 50%;
  }
`;
