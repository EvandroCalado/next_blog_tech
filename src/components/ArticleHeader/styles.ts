import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
