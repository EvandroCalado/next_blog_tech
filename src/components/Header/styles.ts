import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacings.xhuge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.dark};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${Title} {
      margin: ${theme.spacings.small} 0;
      text-align: left;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
