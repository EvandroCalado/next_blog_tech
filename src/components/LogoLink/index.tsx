import * as Styled from './styles';
import Heading from '../Heading';
import Link from 'next/link';

export type LogoLinkProps = {
  text: string;
  image?: string;
  link: string;
  newTab?: boolean;
};

const LogoLink = ({
  text,
  image = '',
  link,
  newTab = false,
}: LogoLinkProps) => {
  const isExternLink = link.match(/^\//) ? true : false;
  const target = newTab ? '_blank' : '_self';

  if (isExternLink) {
    <Heading size="small" uppercase>
      <Link href={link} passHref>
        <Styled.Container target={target}>
          {image ? <img src={image} alt={text} /> : text}
        </Styled.Container>
      </Link>
    </Heading>;
  }

  return (
    <Heading size="small" uppercase>
      <Styled.Container href={link} target={target}>
        {image ? <img src={image} alt={text} /> : text}
      </Styled.Container>
    </Heading>
  );
};

export default LogoLink;
