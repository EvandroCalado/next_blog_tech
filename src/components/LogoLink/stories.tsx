import { Meta, Story } from '@storybook/react/types-6-0';
import LogoLink, { LogoLinkProps } from '.';

export default {
  title: 'LogoLink',
  component: LogoLink,
  args: {
    text: 'LogoLink',
    image:
      'https://res.cloudinary.com/dnhsdenji/image/upload/v1679924887/110628201_ae0753864b.jpg',
    link: 'https://google.com.br',
    newTab: false,
  },
} as Meta;

export const ImageOnly: Story<LogoLinkProps> = (args) => {
  return (
    <div>
      <LogoLink {...args} />
    </div>
  );
};

export const TextOnly: Story<LogoLinkProps> = (args) => {
  return (
    <div>
      <LogoLink {...args} />
    </div>
  );
};

TextOnly.args = {
  image: '',
};
