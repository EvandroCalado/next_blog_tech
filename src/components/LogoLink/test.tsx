import LogoLink from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink link="#target" text="Olá mundo" />);
    expect(
      screen.getByRole('heading', { name: 'Olá mundo' }),
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Olá mundo' })).toHaveAttribute(
      'href',
      '#target',
    );
  });

  it('should render image logo', () => {
    renderTheme(<LogoLink link="#target" text="Olá mundo" image="image.jpg" />);
    expect(screen.getByAltText('Olá mundo')).toHaveAttribute(
      'src',
      'image.jpg',
    );
  });

  it('should render without image logo', () => {
    renderTheme(<LogoLink link="#target" text="Olá mundo" />);
    expect(screen.queryByAltText('Olá mundo')).toBeNull();
  });

  it('should render with link extern', () => {
    renderTheme(<LogoLink link="/target" text="Olá mundo" newTab={true} />);
    expect(
      screen.getByRole('heading', { name: 'Olá mundo' }),
    ).toBeInTheDocument();
  });

  it('should render with link intern', () => {
    renderTheme(<LogoLink link="/target" text="Olá mundo" image="image.jpg" />);
    expect(
      screen.getByRole('heading', { name: 'Olá mundo' }),
    ).toBeInTheDocument();
    expect(screen.getByAltText('Olá mundo')).toHaveAttribute(
      'src',
      'image.jpg',
    );
  });

  it('should render match snapshot', () => {
    renderTheme(<LogoLink link="#target" text="Olá mundo" image="image.jpg" />);
    expect(screen.getByRole('link', { name: 'Olá mundo' })).toMatchSnapshot();
  });
});
