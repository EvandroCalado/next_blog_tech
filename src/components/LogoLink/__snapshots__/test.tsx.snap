// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should render match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 2rem;
}

.c0 > img {
  width: 15rem;
  border-radius: 50%;
}

<a
  class="c0"
  href="#target"
  target="_self"
>
  <img
    alt="OlÃ¡ mundo"
    src="image.jpg"
  />
</a>
`;
